lib/lib.go

```diff
@@ -624,12 +623,20 @@ func NewInstance(ctx context.Context, repoPath string, opts ...Option) (qri *Ins
 	}
 
 	if o.collectionSet == nil && inst.repo != nil {
-		inst.collectionSet, err = collection.NewLocalSet(ctx, inst.bus, repoPath, func(o *collection.LocalSetOptions) {
+		set, err := collection.NewLocalSet(ctx, repoPath, func(o *collection.LocalSetOptions) {
 			o.MigrateRepo = inst.repo
 		})
 		if err != nil {
 			return nil, err
 		}
+		o.collectionSet = set
+	}
+
+	if o.collectionSet != nil {
+		inst.collection, err = collection.NewCollection(ctx, inst.bus, o.collectionSet)
```

Comment:
Ok, I was confused before but I think I can see what's going on now. I'll leave a comment at top-level concerning the biggest thing I think we need to sort out.